coreservices/registration/src/CatenaX.NetworkServices.Registration.Service/Controllers/RegistrationController.cs

```diff
@@ -92,12 +92,24 @@ public async Task<IActionResult> CreateWallet([FromBody] WalletInformation walle
 
         [HttpPost]
         [Route("documents")]
-        [Authorize(Roles = "invite_user")]
+        [Authorize(Roles = "upload_documents")]
         public async Task<IActionResult> CreateDocument([FromForm(Name = "document")] IFormFile document)
         {
-            var username = User.Claims.SingleOrDefault(x => x.Type == "sub").Value as string;
+            try
+            {
+                var username = User.Claims.SingleOrDefault(x => x.Type == "name").Value as string;
```

Comment:
'name' is certainly not a suitable primary key to identify a person in the longrun (neither is 'username' or 'email')
if 'sub' is not present in the claim it's most likely mapped somewhere else by dotnet-standarts. That can easily be disabled, see:
https://github.com/tractusx-team-portal-onboarding/tractusx/blob/a10d5de113c54b9763f2934f97f232031c843f73/coreservices/invitation/src/CatenaX.NetworkServices.Invitation.Service/Startup.cs#L80

Reply:
THanks you for that! 
Sub works now :)